client/factory.go

```diff
@@ -32,19 +30,19 @@ const (
 )
 
 type FactoryConfig struct {
-	CacheDuration time.Duration
-	GSTenantID    string
-	K8sClient     k8sclient.Interface
-	Logger        micrologger.Logger
+	CacheDuration      time.Duration
+	CredentialProvider credential.Provider
+	K8sClient          k8sclient.Interface
+	Logger             micrologger.Logger
 }
 
 // Factory is creating Azure clients for specified AzureConfig CRs, so basically for specified
 // tenant clusters. All created clients are cached.
 type Factory struct {
-	gsTenantID string
-	k8sClient  k8sclient.Interface
-	logger     micrologger.Logger
-	mutex      sync.Mutex
+	credentialProvider credential.Provider
```

Comment:
Instead of directly fetching the credentials, the factory uses the provider. This provider is configured with the GiantSwarm tenant ID.

Reply:
Great idea! I was also thinking about making it more generic, just didn't expect that we will have a need so soon :smile: